client/my-sites/checkout/composite-checkout/hooks/use-remove-from-cart-and-redirect.ts

```diff
@@ -46,8 +51,12 @@ export default function useRemoveFromCartAndRedirect(
 			window.location.href = cartEmptyRedirectUrl;
 			return;
 		}
-		page.redirect( cartEmptyRedirectUrl );
-	}, [ createUserAndSiteBeforeTransaction, siteSlug, siteSlugLoggedOutCart ] );
+		if ( checkoutBackUrl ) {
+			window.location.href = checkoutBackUrl;
```

Comment:
Technically I think this creates an [open redirect](https://cwe.mitre.org/data/definitions/601.html) since we are redirecting the user to a location based on a URL in the query string, which could be anything (unless there's already a guard somewhere before this gets put into state, but I doubt it). I was just talking with someone [about this in another PR](https://github.com/Automattic/wp-calypso/commit/f393dd3051b3f25341cfb635d43b138459d22539#r53059967). That said, I know this is a hard problem to solve; is there any way we can pull the hostname of the Jetpack site from calypso state somehow? If so, we'd be able to compare it to the hostname of `cartEmptyRedirectUrl`.

Reply:
Yes, I see what you're saying about the open redirect.  I'm pretty sure the `checkoutBackUrl` will only be one of 2 values: either `cloud.jetpack.com/pricing/` or the site wp-admin itself. Yes, this is a tricky one because we're not always logged-in at this point, so we don't have `siteId` and therefore we cannot rely on selectors `getSiteName( state, siteId )`, or `getRawSite( state, siteId )`.  But we do have the 'siteSlug' prop passed into checkout. Do you suppose it would be enough to verify that the `checkoutBackUrl` hostname is either `cloud.jetpack.com` or the siteSlug itself that we are checking out on?  Here, I'll commit an example of what I'm talking about..

Reply:
I think (without having tested it) that this looks like a great solution! There's a few other places I'm going to have to revise to do something like this and I'll probably try this same strategy.